File: app/src/main/java/com/xj/anchortask/asyncInflate/ThreadUtils.java
package com.xj.anchortask.asyncInflate;

import android.os.Handler;
import android.os.Looper;

/**
 * Created by jun xu on 4/1/21.
 */
public class ThreadUtils {

    public static void runOnUiThread(Runnable r) {
        if (isMainThread()) {
            r.run();
        } else {
            LazyHolder.sUiThreadHandler.post(r);
        }
    }

    public static void runOnUiThreadAtFront(Runnable r) {
        if (isMainThread()) {
            r.run();
        } else {
            LazyHolder.sUiThreadHandler.postAtFrontOfQueue(r);
        }
    }

    public static void runOnUiThread(Runnable r, long delay) {
        LazyHolder.sUiThreadHandler.postDelayed(r, delay);
    }

    public static void postOnUiThread(Runnable r) {
        LazyHolder.sUiThreadHandler.post(r);
    }

    public static void removeCallbacks(Runnable r) {
        LazyHolder.sUiThreadHandler.removeCallbacks(r);
    }

    public static boolean isMainThread() {
        return Looper.getMainLooper() == Looper.myLooper();
    }

    private static class LazyHolder {
        private static Handler sUiThreadHandler = new Handler(Looper.getMainLooper());
    }
}

File: library/src/main/java/com/xj/anchortask/library/ThreadUtils.kt
package com.xj.anchortask.library

import android.os.Handler
import android.os.Looper

object ThreadUtils {

    fun runOnUiThread(r: Runnable) {
        if (isMainThread) {
            r.run()
        } else {
            LazyHolder.sUiThreadHandler.post(r)
        }
    }

    fun runOnUiThreadAtFront(r: Runnable) {
        if (isMainThread) {
            r.run()
        } else {
            LazyHolder.sUiThreadHandler.postAtFrontOfQueue(r)
        }
    }

    fun runOnUiThread(r: Runnable?, delay: Long) {
        LazyHolder.sUiThreadHandler.postDelayed(r!!, delay)
    }

    fun removeCallbacks(r: Runnable?) {
        LazyHolder.sUiThreadHandler.removeCallbacks(r!!)
    }

    val isMainThread: Boolean
        get() = Looper.getMainLooper() == Looper.myLooper()

    fun checkAtMainThread() {
        val e = RuntimeException("not main thread")
        LazyHolder.sUiThreadHandler.postDelayed({ throw e }, 500)
    }

    private object LazyHolder {
        val sUiThreadHandler =
            Handler(Looper.getMainLooper())
    }
}